client-app/core/api/graphql/catalog/queries/searchRelatedProducts/searchRelatedProducts.graphql

```diff
@@ -25,6 +25,9 @@ query SearchRelatedProducts(
             name
           }
           hasVariations
+          variations {
+            id
+          }
```

Comment:
Do we really need it?

Reply:
> Do we really need it in query?

Yep, we have product.variations?.length in product-card-related

Reply:
Candidate to move into fragments, but for now ok